Measurements/Tests.qs

```diff
@@ -36,18 +37,28 @@ namespace Quantum.Kata.Measurements {
                 // do state prep: convert |0⟩ to outcome with false return or to outcome with true return depending on state
                 statePrep(qs[0], state);
 
-                // get the solution's answer and verify that it's a match
+                // get the solution's answer and verify if NOT a match, then differentiate what kind of mismatch
                 let ans = testImpl(qs[0]);
-                if (ans == (state == 1)) {
-                    set nOk += 1;
+                if (ans != (state == 1)) {
+                    if (state == 1) {
+                        set nNOTOk1as0 += 1;
+                    }
+                    else {
+                        set nNOTOk0as1 += 1;
+                    }
                 }
 
                 // we're not checking the state of the qubit after the operation
                 Reset(qs[0]);
             }
         }
 
-        EqualityFactI(nOk, nTotal, $"{nTotal - nOk} test runs out of {nTotal} returned incorrect state.");
+        // This Check will tell the # of Failed clasifications
+        // EqualityFactI((nNOTOk0as1 + nNOTOk1as0), 0, $"{nNOTOk0as1 + nNOTOk1as0} test runs out of {nTotal} returned incorrect state.");
+        
+        // This Check will give more granularity telling the # of misclasifications by type (0->1) vs (1->0)
+        // Gets two strings as input of what are the two states to be clasified, str_aim being the target state. 
+        EqualityFactI((nNOTOk0as1 + nNOTOk1as0), 0, $"'Misclassified' {str1} as {str_aim} in {nNOTOk0as1} test runs, and {str_aim} as {str1} in {nNOTOk1as0} test runs out of {nTotal} total runs.");
```

Comment:
I would split this check in two parts (having in mind that the later tasks will need to report errors on more states): 
* first go through all possible misclassifications (in this case iterate over the array) and report to output (using `Message`) only those that actually happened (for example, if the code always returns "|0⟩" state, we won't report "|0⟩ was misclassified as |1⟩ 0 times", we'll only point the actual mistakes done). At the same time we'd accumulate the total count of misclassifications during this iteration.
* after that do a `fail` if we have any misclassifications, saying something like "X test runs out of Y returned incorrect state (see output for details)".

Reply:
Makes sense. The detailed output will look like:

'Misclassified' |0⟩ as |1⟩ in 32 test runs out of 100 total runs.
'Misclassified' |1⟩ as |0⟩ in 22 test runs out of 100 total runs.
54 ≠ 0: 54 test runs out of 100 returned incorrect state (see output for details).
Try again!